source/code/plugins/agent_topology_request_script.rb

```diff
@@ -0,0 +1,143 @@
+require 'optparse'
+require 'gyoku'
+
+class StrongTypedClass
+  def self.strongtyped_accessor(name, type)
+    # setter
+    self.class_eval("def #{name}=(value);
+      if !value.is_a? #{type} and !value.nil?
+        raise ArgumentError, \"Invalid data type. #{name} should be type #{type}\"
+      end
+      @#{name}=value
+    end")
+    
+    # getter
+    self.class_eval("def #{name};@#{name};end")
+  end
+
+  def self.strongtyped_arch(name)
+    # setter
+    self.class_eval("def #{name}=(value);
+      if (value != 'x64' && value != 'x86')
+        raise ArgumentError, \"Invalid data for ProcessorArchitecture.\"
+      end
+      @#{name}=value
+    end")
+  end
+end
+
+class AgentTopologyRequestOperatingSystemTelemetry < StrongTypedClass
+  strongtyped_accessor :PercentUserTime, Integer
```

Comment:
If they are nil or not assigned, telemetry will be nil, but OperatingSystem  info will still go through to the topology request